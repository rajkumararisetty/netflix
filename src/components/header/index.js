import { useState } from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Background,
  Feature,
  Text,
  TextLink,
  Container,
  Logo,
  ButtonLink,
  FeatureCallOut,
  Group,
  Dropdown,
  Profile,
  Picture,
  Search,
  SearchIcon,
  SearchInput,
  PlayButton,
} from "./styles/header";

export default function Header({ children, bg = true, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <TextLink {...restProps}>{children}</TextLink>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`images/users/${src}.png`} />;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);
  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((searchActive) => !searchActive)}
      >
        <img src="/images/icons/search.png" alt="search" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search firms and series"
        active={searchActive}
      />
    </Search>
  );
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};
