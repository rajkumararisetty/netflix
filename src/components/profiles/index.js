import { Container, Title, List, Item, Picture, Name } from "./styles/profiles";

export default function Profiles({ children, ...resetProps }) {
  return <Container {...resetProps}>{children}</Container>;
}

Profiles.Title = function ProfilesTitle({ children, ...resetProps }) {
  return <Title {...resetProps}>{children}</Title>;
};

Profiles.List = function ProfilesList({ children, ...resetProps }) {
  return <List {...resetProps}>{children}</List>;
};

Profiles.Item = function ProfilesItem({ children, ...resetProps }) {
  return <Item {...resetProps}>{children}</Item>;
};

Profiles.Picture = function ProfilesPicture({ src, ...resetProps }) {
  return (
    <Picture
      {...resetProps}
      src={src ? `/images/users/${src}.png` : `/images/misc/loading.gif`}
    />
  );
};

Profiles.Name = function ProfilesName({ children, ...resetProps }) {
  return <Name {...resetProps}>{children}</Name>;
};
