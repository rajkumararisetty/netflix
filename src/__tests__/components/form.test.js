import { Form } from "../../components";
import { render } from '@testing-library/react';

jest.mock('react-router-dom', () => {
	return {
		Link: props => {
			return <div {...props}></div>;
		},
	}
});

describe("<Form />", () => {
  it("renders the <Form /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign In Now</Form.Title>
        <Form.Base>
          <Form.Input placeholder="Email Address" onChange={() => {}} />
          <Form.Input
            placeholder="Password"
            type="password"
            onChange={() => {}}
          />
          <Form.Submit disabled type="submit">
            Sign In
          </Form.Submit>
        </Form.Base>
        <Form.Text>
          New to Netflix? <Form.Link to="/signup">Sign up now.</Form.Link>
        </Form.Text>
        <Form.TextSmall>
          This page is protected by google reCAPTCHA to ensure you're not a bot.
          Learn more.
        </Form.TextSmall>
      </Form>
	);
	
	expect(
		getByText("This page is protected by google reCAPTCHA to ensure you're not a bot. Learn more.")
	).toBeTruthy();

	expect(getByText('Sign In Now')).toBeTruthy();
	expect(getByText('Sign In')).toBeTruthy();
	expect(getByText('Sign In').disabled).toBeTruthy();
	expect(getByPlaceholderText('Email Address')).toBeTruthy();
	expect(getByPlaceholderText('Password')).toBeTruthy();
	expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Form /> with an error', () => {
	  const { container, getByText, queryByText } = render(
		  <Form>
			  <Form.Error>Your email address is already being used</Form.Error>
			  <Form.Submit type="submit">Sign In</Form.Submit>
		  </Form>
	  );

	  expect(getByText('Your email address is already being used')).toBeTruthy();
	  expect(queryByText('Sign In').disabled).toBeFalsy();
	  expect(container.firstChild).toMatchSnapshot();
  });
});
