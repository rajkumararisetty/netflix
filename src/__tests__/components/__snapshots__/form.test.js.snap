// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with an error 1`] = `
<div
  class="form__Container-sc-7952rh-0 cLQAcv"
>
  <div
    class="form__Error-sc-7952rh-2 hWjwPU"
  >
    Your email address is already being used
  </div>
  <button
    class="form__Submit-sc-7952rh-8 eMdJKv"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<Form /> renders the <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-7952rh-0 cLQAcv"
>
  <h1
    class="form__Title-sc-7952rh-3 dWhmiH"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-7952rh-1 bZCFBJ"
  >
    <input
      class="form__Input-sc-7952rh-7 fJWbJd"
      placeholder="Email Address"
    />
    <input
      class="form__Input-sc-7952rh-7 fJWbJd"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-7952rh-8 eMdJKv"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-7952rh-4 fvOBTc"
  >
    New to Netflix? 
    <div
      class="form__Link-sc-7952rh-6 cplCKY"
      to="/signup"
    >
      Sign up now.
    </div>
  </p>
  <p
    class="form__TextSmall-sc-7952rh-5 drVMew"
  >
    This page is protected by google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;
